### PR TITLE
feat: add lint-staged and prettier

### PR DESCRIPTION
**Pull request by ojeytonwilliams**
This should keep the noise down a little without being too annoying.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -37,12 +38,23 @@
...
     }
-  ]
+  ],
+  "husky": {
+    "hooks": {
+      "pre-commit": "lint-staged"
+    }
+  },
+  "lint-staged": {
+    "**/*.js": "prettier --write"
```
Comment (ShaunSHamilton) [inline package.json:58]:
```suggestion
    "**/*.js": "prettier --write ."
```

**Review by ShaunSHamilton — COMMENTED** [review]
Question: Why is this not?

File: package.json (hunk 1 above)
Comment (ojeytonwilliams) [reply package.json:58]: Good question. So, what happens here is as follows:

Each file that matches the glob `**/*.js` gets passed into `prettier --write` as an argument e.g. `/lib/index.js` gets picked up and lint-staged will run `prettier --write /lib/index.js`.

Comment (ShaunSHamilton) [reply package.json:58]: Ah. Very clever. Thank you, for clearing that up.

**Pull request merged**